### ISSUE TITLE
no-unused-vars should not trigger array spread

### ISSUE THREAD

**Issue by piuccio** [eslint/eslint#9598]
* **ESLint Version:** 4.10.0

With the following code

```js
const [one, two, three] = [1, 2, 3];
console.log(three);
```

I get the error that

```
'one' is assigned a value but never used
'two' is assigned a value but never used
```

I'm expecting that nothing happens because when I spread an array, the order is important.

I tried 

```
"no-unused-vars": [2, {"ignoreRestSiblings": true}],
```

but it makes no difference because I'm not using the rest operator.

I'm aware I can use

```js
const [,, three] = [1, 2, 3];
```

but I think this syntax is quite error prone and confusing.

Would be great to have an option to turn off this behavior similarly to what `{ "args": "after-used" }` does for function arguments

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
